keps/sig-api-machinery/2144-clientgo-apply/README.md

```diff
@@ -221,59 +220,17 @@ While there are arguments to be made all of these fields are expected to cause
 problems in practice, there are some that clearly would, e.g. ContainerPort.
 
 Because of this we cannot use the existing go structs to represent apply
-configurations.
+configurations. Instead, we will generated "builders".
 
-<<[UNRESOLVED @jpbetz @jennybuckley ]>> 
-Finalize which alternative to use based on developer feedback. See the
-[Alternatives](#alternatives) for a complete list, but are currently focusing on
-the two below alternatives. We are working with the Kubebuilder community to
-gather feedback on what developers prefer.
-<<[/UNRESOLVED]>>
-
-#### Alternative 1: Generated structs where all fields are pointers
-
-Example usage:
-
-```go
-import (
-  ptr "k8s.io/utils/pointer"
-)
-
-&appsv1apply.Deployment{
-  Name: ptr.StringPtr("nginx-deployment"),
-  Spec: &appsv1apply.DeploymentSpec{
-    Replicas: ptr.Int32Ptr(0),
-    Template: &v1apply.PodTemplate{
-      Spec: &v1apply.PodSpec{
-        Containers: []v1.Containers{
-          {
-            Name: ptr.StringPtr("nginx"),
-            Image: ptr.StringPtr("nginx:latest"),
-          },
-        }
-      },
-    },
-  },
-}
-```
-
-For this approach, developers need to use a library like
-https://github.com/kubernetes/utils/blob/master/pointer/pointer.go
-to inline primitive literals.
-
-#### Alternative 2: Generated "builders"
-
-Example usage:
+Example generated builder:
 
 ```go
-&appsv1apply.Deployment().
-  ObjectMeta(&metav1apply.ObjectMeta().
-    Name("nginx-deployment")).
-  Spec(&appsv1apply.DeploymentSpec().
+&appsv1apply.Deployment("ns", "nginx-deployment").
```

Comment:
You still have an extra ampersand here ;-) But the mandatory name is good!

Reply:
This is what happens when I hand write examples in .md files :)

I fixed the Containers() function too. Now it is varadic.